datasync/chaindatafetcher/kafka/kafka.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/hashicorp/go-uuid"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/types"
+	"github.com/klaytn/klaytn/log"
+)
+
+var (
+	logger = log.NewModuleLogger(log.ChainDataFetcher)
+	kb     *KafkaBroker
+	once   sync.Once
+)
+
+type KafkaBroker struct {
```

Comment:
```suggestion
type Kafka struct {
```
I think just `Kafka` looks better for abstraction. It is up to you actually. :)

Reply:
Agreed! 55a975720c3097a2b2108b666979ae02da2b30a7